### PR TITLE
Update upgrading-talos.

### PR DESCRIPTION
**Pull request by steverfrancis**
Variety of clarifications.

included the good bits from https://www.talos.dev/docs/v0.14/learn-more/upgrades/
(Which can be deleted now, but I dont know how to do :-) ).

# Pull Request

## Acceptance

Please use the following checklist:

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [ ] you ran conformance (`make conformance`)
- [ ] you formatted your code (`make fmt`)
- [ ] you linted your code (`make lint`)
- [ ] you generated documentation (`make docs`)
- [ ] you ran unit-tests (`make unit-tests`)

> See `make help` for a description of the available targets.

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/4907)

### CODE REVIEW

File: website/content/docs/v0.15/Guides/upgrading-talos.md [hunk 1]
```diff
@@ -57,3 +62,71 @@ future.
...
+From the user's standpoint, however, the processes are identical. However, since control plane nodes run additional services, such as etcd, there are some extra steps and checks performed on them. 
+For instance, Talos will refuse to upgrade a control plane node if that upgrade would cause a loss of quorum for etcd.
+If running a single-node cluster, and you want to force an upgrade despite the loss of quorum, you can set `preserve` to `true`. 
+
+**Q.** Will an upgrade try to do the whole cluster at once?
+Can I break my cluster by upgrading everything?
+
+**A.** No.
+
+Nothing prevents the user from sending near-simultaneous upgrades to each node of the cluster.
```
Comment (smira) [inline website/content/docs/v0.15/Guides/upgrading-talos.md:127]: I think this needs further clarification, as upgrading every node at the same time might break stateful applications like Rook/Ceph. I wouldn't recommend during simultaneous upgrades in the docs. Even though it's possible to do such upgrades, I feel we shouldn't recommend that.

File: website/content/docs/v0.15/Guides/upgrading-talos.md [hunk 2]
```diff
@@ -57,3 +62,71 @@ future.
...
+Lastly, it is possible that Talos itself will upgrade successfully, start up, and rejoin the cluster but your workload will fail to run on it, for whatever reason.
+This is when you would use the `talosctl rollback` command to revert back to the previous Talos version.
+
+**Q.** Can upgrades be scheduled?
+
+**A.** Because the upgrade sequence is API-driven, you can easily tie it in to your own business logic to schedule and coordinate your upgrades.
+
+**Q.** Can the upgrade process be observed?
+
+**A.** Yes, using the `talosctl logs --follow machined` command.
```
Comment (smira) [inline website/content/docs/v0.15/Guides/upgrading-talos.md:111]: should be `talosctl dmesg -f`

Comment (steverfrancis) [reply website/content/docs/v0.15/Guides/upgrading-talos.md]: Updated

File: website/content/docs/v0.15/Guides/upgrading-talos.md (hunk 1 above)
Comment (steverfrancis) [reply website/content/docs/v0.15/Guides/upgrading-talos.md]: Clarificaiton added. (I just copy/pasted this from the other doc, so glad you are reviewing this!)

**Pull request closed without merging**